etna/transforms/holiday.py

```diff
@@ -0,0 +1,66 @@
+import datetime
+from typing import Optional
+
+import holidays
+import numpy as np
+import pandas as pd
+
+from etna.transforms.base import Transform
+
+
+class HolidayTransform(Transform):
+    """HolidayTransform generates series that indicates holidays in given dataframe. Creates column 'regressor_holidays'."""
+
+    def __init__(self, iso_code: str = "RUS", out_column: Optional[str] = None):
+        """
+        Create instance of HolidayTransform.
+        Parameters
+        ----------
+        iso_code:
+            internationally recognised codes, designated to country for which we want to find the holidays
+        out_column:
+            name of added column. Use self.__repr__() if not given.
+        """
+        self.iso_code = iso_code
+        self.holidays = holidays.CountryHoliday(iso_code)
+        self.out_column = out_column
+        self.out_column = self.out_column if self.out_column is not None else f"regressor_{self.__repr__()}"
+
+    def fit(self, df: pd.DataFrame) -> "HolidayTransform":
+        """
+        Fit HolidayTransform with data from df. Does nothing in this case.
+        Parameters
+        ----------
+        df: pd.DataFrame
+            value series with index column in timestamp format
+        """
+        return self
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transform data from df with HolidayTransform and generate a column of holidays flags.
+        Parameters
+        ----------
+        df: pd.DataFrame
+            value series with index column in timestamp format
+        Returns
+        -------
+            pd.DataFrame with 'regressor_holidays' column
+        """
+        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
+            raise ValueError("Frequency of data should be no more than daily.")
```

Comment:
why do we need this check?

Reply:
From the issue statement: 'It will work with daily frequency and less (hourly, minutely, etc.)'